src/client/client.ts

```diff
@@ -0,0 +1,76 @@
+import { FetchParams, ICache } from '../types'
+import fetch from 'node-fetch'
+
+/**
+ * A helper wrapper to access the running Authless-Server
+ * and fetch the response {@link IResponse}
+ *
+ * @remarks
+ * Can use a cache {@link ICache} if available and passed to constructor
+ */
+export class AuthlessClient {
+  cache?: ICache
+
+  /**
+   * Takes an optional Cache as a parameter which should satisfy {@link ICache}
+   */
+  constructor (cache?: ICache) {
+    this.cache = cache
+  }
+
+  /**
+   * Makes a HTTP GET query from the passed URLParams {@link URLParams}
+   *
+   * @param params - Object from which to generate a stringified HTTP Request query
+   */
+  static makeParams (params: FetchParams): string {
+    let retParams = `u=${params.url}`
+    retParams += ['responseFormat', 'inputs', 'alphabetSelector', 'referer', 'username'].reduce((acc, key) => {
+      const param = params[key]
+      if(typeof param === 'string') {
+        return `${acc}&${key}=${encodeURI(param)}`
+      }
+      return acc
+    }, '')
+    return retParams
+  }
+
+  /**
+   * Fetch the response from the running Authless-Server
+   *
+   * @remarks
+   * Will return a cached value if possible dependind on the cache passed
+   * to the AuthlessClient constructor
+   *
+   * @param params - Parameters to access the Authless-Server and scrape the target {@link FetchParams}
+   */
+  async fetch (params: FetchParams): Promise<any> {
```

Comment:
I want to see more tsdocs for this method. At least one example and a full list of what params are available.